### PR TITLE
Enhance vapp network with dhcp capabilities

### PR DESCRIPTION
**Pull request by vbauzysvmware**
Reference: terraform-providers/terraform-provider-vcd#97

Existing API enhanced to support ability to configure DHCP. In UI functionality is visible selecting vapp network and "configure services...".

### CODE REVIEW

File: govcd/vapp.go [hunk 1]
```diff
@@ -1097,17 +1110,54 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
 
 }
 
+func validateNetworkConfigSettings(networkSettings *VappNetworkSettings) error {
+	if networkSettings.Name == "" {
+		return errors.New("network name is missing")
+	}
+
+	if networkSettings.Gateway == "" {
+		return errors.New("network gateway ip is missing")
```
Comment (dataclouder) [inline govcd/vapp.go]: Please change "ip" to "IP" when it is inside text messages.

File: govcd/vapp.go [hunk 2]
```diff
@@ -1097,17 +1110,54 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
...
+	isNetworkFound := false
 	for index, networkConfig := range networkConfigurations {
 		if networkConfig.NetworkName == networkName {
+			isNetworkFound = true
 			networkConfigurations = append(networkConfigurations[:index], networkConfigurations[index+1:]...)
 		}
 	}
 
+	if !isNetworkFound {
+		return Task{}, fmt.Errorf("network to remove: %s, isn't found", networkName)
```
Comment (dataclouder) [inline govcd/vapp.go]: "network to remove: %s, isn't found" -> "network to remove %s, wasn't found"

File: govcd/vapp.go (hunk 1 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: Fixed

File: govcd/vapp.go (hunk 2 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: Fixed

**Pull request merged**